server/src/main/java/io/crate/analyze/AnalyzedColumnDefinition.java

```diff
@@ -363,12 +363,14 @@ public static void validateAndComputeDocValues(AnalyzedColumnDefinition<Object>
                     DataType<?> dataType = definition.dataType();
                     boolean val = storageSettings.getAsBoolean(property, true);
                     if (val == false) {
-                        if (dataType.id() != DataTypes.STRING.id()) {
+                        if (DataTypes.isNumericPrimitive(dataType) || DataTypes.STRING.id() == dataType.id()) {
+                            definition.docValues = false;
+                        } else {
                             throw new IllegalArgumentException(
-                                String.format(Locale.ENGLISH, "Invalid storage option \"columnstore\" for data type \"%s\"",
-                                    dataType.getName()));
+                                String.format(Locale.ENGLISH,
+                                              "Invalid storage option \"columnstore\" for data type \"%s\"",
+                                              dataType.getName()));
                         }
-                        definition.docValues = false;
```

Comment:
Could we extend `StorageSupport` with a new property and generify this, so a `AnalyzedColumnDefinition` isn't aware of whether a type supports doc values or not?

It would avoid spreading custom logic in regards to types across the code base as it makes it much more difficult to add new types and we may want some kind of user-defined-type or TypePlugin support at some point in the future. (There are exceptions to this where the types are already special anyways, like arrays and objects)

Reply:
good idea, please check again.